DSLink.Tests/ResponderTests.cs

```diff
@@ -0,0 +1,231 @@
+                    {
+                        new JObject
+                        {
+                            new JProperty("path", "/testValue"),
+                            new JProperty("sid", 0)
+                        }
+                    })
+                }
+            });
+        }
+
+        private Task<JArray> _unsubscribeFromNode()
+        {
+            return _responder.ProcessRequests(new JArray
+            {
+                new JObject
+                {
+                    new JProperty("rid", 2),
+                    new JProperty("method", "unsubscribe"),
+                    new JProperty("sids", new JArray
+                    {
+                        0
+                    })
+                }
+            });
+        }
+
+        [Test]
+        public async Task TestList()
+        {
+            var responses = await _listNode();
+            var response = responses[0];
+            var updates = response["updates"];
+
+            Assert.AreEqual(1, response["rid"].Value<int>());
+            Assert.AreEqual("open", response["stream"].Value<string>());
+
+            Assert.IsTrue(JToken.DeepEquals(updates[0], new JArray
+            {
+                "$is",
+                "node"
+            }));
+
+            var testValueUpdate = updates[1][1];
+            Assert.AreEqual("testValue", updates[1][0].Value<string>());
+            Assert.NotNull(testValueUpdate["$is"]);
+            Assert.NotNull(testValueUpdate["$type"]);
+            Assert.NotNull(testValueUpdate["value"]);
+            Assert.NotNull(testValueUpdate["ts"]);
+            Assert.AreEqual("node", testValueUpdate["$is"].Value<string>());
+            Assert.AreEqual("number", testValueUpdate["$type"].Value<string>());
+            Assert.AreEqual(123, testValueUpdate["value"].Value<int>());
+            Assert.AreEqual(JTokenType.String, testValueUpdate["ts"].Type);
+
+            var testNodeUpdate = updates[2][1];
+            Assert.AreEqual("testNodeConfigs", updates[2][0].Value<string>());
+            Assert.NotNull(testNodeUpdate["$is"]);
+            Assert.NotNull(testNodeUpdate["$testString"]);
+            Assert.NotNull(testNodeUpdate["$testNumber"]);
+            Assert.AreEqual("node", testNodeUpdate["$is"].Value<string>());
+            Assert.AreEqual("string", testNodeUpdate["$testString"].Value<string>());
+            Assert.AreEqual(123, testNodeUpdate["$testNumber"].Value<int>());
+
+            Console.WriteLine(updates.ToString());
+        }
+
+        [Test]
+        public async Task TestInvoke()
+        {
+            bool actionInvoked = false;
+            _responder.SuperRoot.CreateChild("testAction")
+                .SetInvokable(Permission.Write)
+                .SetAction(new ActionHandler(Permission.Write, async (request) =>
+                {
+                    Assert.AreEqual("string", request.Parameters["testString"].Value<string>());
```

Comment:
This could be split into 2 tests. One checking that the wiring of the invocation actually works, and another one to make sure the parameters are well bound.